### DISCUSSION TITLE
Using Redis cache slowed down Directus

### DISCUSSION THREAD

**Discussion by AbraamNader** [directus/directus#6995] [Feature Requests]
After deploying Directus on a high traffic website with a lot of concurrent users, Directus slowed down a lot, but one of the things that helped (at least for a while) was flushing all the redis keys, has that happened to anyone else? perhaps too many keys were created?

Also does Directus have a fallback if redis was not responsive?

Note: 
This option was enabled `CACHE_AUTO_PURGE=true`

**Comment by rijkvanzanten** [comment]
Interesting! I haven't heard of this problem before. Are you doing a lot of writes? The auto-purge setting will flush the cache on changes of content, which requires the caches to be recreated. If you're doing a lot of writes, this might slow down the API a bit, as it's creating fresh cache records more often 🤔

**Reply by AbraamNader** [reply]
Not that many, our content team that is currently using Directus is not big yet and the data is not updated very frequently, since this is the first time we used Directus in production, however our public userbase is big, and all they do is just reads.

I think it could be a redis related issue, rather than a Directus issue, its an external component so it can go down or become unresponsive for any reason, that is why I asked about a fallback.

**Comment by rijkvanzanten** [comment]
> Also does Directus have a fallback if redis was not responsive?

Not at this time. I'd love to hear your thoughts on how that should work 👍🏻

**Reply by AbraamNader** [reply]
I was thinking 
Directus, can check if Redis is healthy (Timeout?), 
if not it would either use memory cache
or just re-execute the query in the DB.

also maybe something in the admin app to reflect that the redis instance is down?

**Reply by rijkvanzanten** [reply]
Good one! It doesn't currently, but we should be able to add try/catch around here to handle the case when the cache fails to extract the data:

https://github.com/directus/directus/blob/ed0029178f7c1dfd3ec5a25e12b1fa5edcb9879a/api/src/middleware/cache.ts#L21

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
